src/topology/algebra/matrix.lean

```diff
@@ -13,31 +13,180 @@ This file is a place to collect topological results about matrices.
 
 ## Main definitions:
 
- * `continuous_det`: the determinant is continuous over a topological ring.
+ * `matrix.topological_ring`: square matrices form a topological ring
+ * `continuous.matrix_det`: the determinant is continuous over a topological ring.
+ * `continuous.matrix_adjugate`: the adjugate is continuous over a topological ring.
 -/
 
 open matrix
+open_locale matrix
 
-variables {ι k : Type*} [topological_space k]
+variables {X α l m n p S R : Type*} [topological_space X] [topological_space R]
 
-instance : topological_space (matrix ι ι k) := Pi.topological_space
+instance : topological_space (matrix m n R) := Pi.topological_space
 
-variables [fintype ι] [decidable_eq ι] [comm_ring k] [topological_ring k]
+instance [has_scalar α R] [has_continuous_const_smul α R] :
+  has_continuous_const_smul α (matrix n n R) :=
+pi.has_continuous_const_smul
 
-lemma continuous_det : continuous (det : matrix ι ι k → k) :=
+instance [topological_space α] [has_scalar α R] [has_continuous_smul α R] :
+  has_continuous_smul α (matrix n n R) :=
+pi.has_continuous_smul
+
+/-- To show a function into matrices is continuous it suffices to show the coefficients of the
+resulting matrix are continuous -/
+@[continuity]
+lemma continuous_matrix [topological_space α] {f : α → matrix m n R}
+  (h : ∀ i j, continuous (λ a, f a i j)) : continuous f :=
+continuous_pi $ λ _, continuous_pi $ λ j, h _ _
+
+lemma continuous.matrix_elem {A : X → matrix m n R} (hA : continuous A) (i : m) (j : n) :
+  continuous (λ x, A x i j) :=
+(continuous_apply_apply i j).comp hA
+
+@[continuity]
+lemma continuous.matrix_map [topological_space S] {A : X → matrix n m S} {f : S → R}
+   (hA : continuous A) (hf : continuous f) :
+  continuous (λ x, (A x).map f) :=
+continuous_matrix $ λ i j, hf.comp $ hA.matrix_elem _ _
+
+@[continuity]
+lemma continuous.matrix_transpose {A : X → matrix n m R} (hA : continuous A) :
+  continuous (λ x, (A x)ᵀ) :=
+continuous_matrix $ λ i j, hA.matrix_elem j i
+
+/-! TODO: add a `has_continuous_star` typeclass so we can write
+```
+lemma continuous.matrix.conj_transpose [has_star R] {A : X → matrix n m R} (hA : continuous A) :
+  continuous (λ x, (A x)ᴴ) :=
+hA.matrix_transpose.matrix_map continuous_star
+```
+-/
+
+@[continuity]
+lemma continuous.matrix_col {A : X → n → R} (hA : continuous A) : continuous (λ x, col (A x)) :=
+continuous_matrix $ λ i j, (continuous_apply _).comp hA
+
+@[continuity]
+lemma continuous.matrix_row {A : X → n → R} (hA : continuous A) : continuous (λ x, row (A x)) :=
+continuous_matrix $ λ i j, (continuous_apply _).comp hA
+
+@[continuity]
+lemma continuous_matrix.diagonal [has_zero R] [decidable_eq n] {A : X → n → R} (hA : continuous A) :
+  continuous (λ x, diagonal (A x)) :=
+continuous_matrix $ λ i j, begin
+  obtain rfl | hij := decidable.eq_or_ne i j,
+  { simp_rw diagonal_apply_eq,
+    exact (continuous_apply _).comp hA },
+  { simp_rw diagonal_apply_ne hij,
+    exact continuous_zero },
+end
```

Comment:
Do you want to write a lemma `continuous.if_prop` which is like `continuous.if` but where `p` doesn't depend on the variable? If so, this could be something like `continuous_matrix $ λ i j, ((continuous_apply _).comp hA).if_prop continuous_zero`.

Reply:
I went for `if_const`, but can use `if_prop` if you feel strongly.

Reply:
No, that is also fine by me. I was a little worried that it sounds like one of the functions is constant instead of the predicate, but it seems fine.